### PR TITLE
Add Fortran version of defaultmap test plus C version cleanup

### PR DESCRIPTION
**Pull request by jhdavis8**
This is a Fortran version of `test_target_teams_distribute_defaultmap`. It checks for mapping of all availible scalar types, so it is quite long.

At present it fails to compile on gfortran with the message `lto1: fatal error: unsupported mode TF`, and passes xlf.

### CODE REVIEW

File: tests/4.5/target_teams_distribute/test_target_teams_distribute_defaultmap.F90 [hunk 1]
```diff
@@ -0,0 +1,422 @@
...
+    INTEGER(kind = 2) :: scalar_short
+    INTEGER(kind = 4) :: scalar_int
+    INTEGER(kind = 8) :: scalar_long_int
+    REAL(kind = 4) :: scalar_float
+    REAL(kind = 8) :: scalar_double
+    LOGICAL :: scalar_logical
+    LOGICAL(kind = 4) :: scalar_logical_kind4
+    LOGICAL(kind = 8) :: scalar_logical_kind8
+    COMPLEX :: scalar_complex
+    COMPLEX(kind = 16) :: scalar_double_complex
```
Comment (josemonsalve2) [inline tests/4.5/target_teams_distribute/test_target_teams_distribute_defaultmap.F90]: See issue #41 . We had to remove this complex kind

File: tests/4.5/target_teams_distribute/test_target_teams_distribute_defaultmap.F90 [hunk 2]
```diff
@@ -0,0 +1,422 @@
...
+    INTEGER(kind = 2),DIMENSION(N) :: short_array
+    INTEGER(kind = 4),DIMENSION(N) :: int_array
+    INTEGER(kind = 8),DIMENSION(N) :: long_int_array
+    REAL(kind = 4),DIMENSION(N) :: float_array
+    REAL(kind = 8),DIMENSION(N) :: double_array
+    LOGICAL,DIMENSION(N) :: logical_array
+    LOGICAL(kind = 4),DIMENSION(N) :: logical_kind4_array
+    LOGICAL(kind = 8),DIMENSION(N) :: logical_kind8_array
+    COMPLEX,DIMENSION(N) :: complex_array
+    COMPLEX(kind = 16),DIMENSION(N) :: double_complex_array
```
Comment (josemonsalve2) [inline tests/4.5/target_teams_distribute/test_target_teams_distribute_defaultmap.F90]: Same here

File: tests/4.5/target_teams_distribute/test_target_teams_distribute_defaultmap.F90 [hunk 3]
```diff
@@ -0,0 +1,422 @@
...
+    INTEGER(kind = 8) :: scalar_long_int
+    REAL(kind = 4) :: scalar_float
+    REAL(kind = 8) :: scalar_double
+    LOGICAL :: scalar_logical
+    LOGICAL(kind = 4) :: scalar_logical_kind4
+    LOGICAL(kind = 8) :: scalar_logical_kind8
+    COMPLEX :: scalar_complex
+    COMPLEX(kind = 16) :: scalar_double_complex
+
+    CHARACTER,DIMENSION(N) :: char_array
```
Comment (josemonsalve2) [inline tests/4.5/target_teams_distribute/test_target_teams_distribute_defaultmap.F90]: Also, It was also discussed in Issue #41 that Characters are not considered scalar for Fortran: 
Section 1.2.6, page 11:
For Fortran: A scalar variable with intrinsic type, as defined by the base language,
excluding character type.

File: tests/4.5/target_teams_distribute/test_target_teams_distribute_defaultmap.F90 [hunk 4]
```diff
@@ -0,0 +1,422 @@
...
+       scalar_short = 83
+       scalar_int = 49
+       scalar_long_int = 12345
+       scalar_float  = 11.2
+       scalar_double  = 9.5
+       scalar_logical = .false.
+       scalar_logical_kind4 = .false.
+       scalar_logical_kind8 = .false.
+       scalar_complex = (5, 5)
+       scalar_double_complex = (5e+9, 5e+9)
```
Comment (josemonsalve2) [inline tests/4.5/target_teams_distribute/test_target_teams_distribute_defaultmap.F90]: Data races?

File: tests/4.5/target_teams_distribute/test_target_teams_distribute_defaultmap.F90 [hunk 5]
```diff
@@ -0,0 +1,422 @@
...
+    OMPVV_TEST_AND_SET_VERBOSE(errors, ABS(IMAG(scalar_complex) - 5) .gt. .00001)
+    OMPVV_TEST_AND_SET_VERBOSE(errors, ABS(REAL(scalar_double_complex) - 5e+9) .gt. .0000000001)
+    OMPVV_TEST_AND_SET_VERBOSE(errors, ABS(IMAG(scalar_double_complex) - 5e+9) .gt. .0000000001)
+
+    test_defaultmap_on = errors
+  END FUNCTION test_defaultmap_on
+
+  INTEGER FUNCTION test_defaultmap_off()
+    INTEGER :: errors, x, y
+    CHARACTER :: scalar_char, scalar_char_copy
```
Comment (josemonsalve2) [inline tests/4.5/target_teams_distribute/test_target_teams_distribute_defaultmap.F90]: Same characters problem

File: tests/4.5/target_teams_distribute/test_target_teams_distribute_defaultmap.F90 [hunk 6]
```diff
@@ -0,0 +1,422 @@
...
+    INTEGER(kind = 4) :: scalar_int, scalar_int_copy
+    INTEGER(kind = 8) :: scalar_long_int, scalar_long_int_copy
+    REAL(kind = 4) :: scalar_float, scalar_float_copy
+    REAL(kind = 8) :: scalar_double, scalar_double_copy
+    LOGICAL :: scalar_logical, scalar_logical_copy
+    LOGICAL(kind = 4) :: scalar_logical_kind4, scalar_logical_kind4_copy
+    LOGICAL(kind = 8) :: scalar_logical_kind8, scalar_logical_kind8_copy
+    COMPLEX :: scalar_complex, scalar_complex_copy
+    COMPLEX(kind = 16) :: scalar_double_complex
+    COMPLEX(kind = 16) :: scalar_double_complex_copy
```
Comment (josemonsalve2) [inline tests/4.5/target_teams_distribute/test_target_teams_distribute_defaultmap.F90]: Same kind 16 problem

**Comment by jhdavis8** [comment]
Added the protection against data races to `_parallel_for_defaultmap.c`. the changes caused gcc to start failing the test, we suspect this is a bug.

**Comment by jhdavis8** [comment]
GCC bug report
https://gcc.gnu.org/bugzilla/show_bug.cgi?id=93515

**Comment by jhdavis8** [comment]
Passing all compilers, other than the reported gcc bug.

**Pull request merged**